spec/controllers/blocks_controller_spec.rb

```diff
@@ -4,17 +4,33 @@
 
 describe BlocksController, type: :request do
   describe '#destroy' do
-    let!(:block) { create(:block) }
+    let(:course) { create(:course) }
+    let!(:week) { create(:week, course_id: course.id) }
+    let!(:block) { create(:block, week_id: week.id)}
     let(:admin) { create(:admin, id: 2) }
-
+    let (:course_approved) do
+      course.campaigns << Campaign.first
+    end 
     before do
       allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(admin)
     end
 
-    it 'destroys the block' do
-      expect(Block.count).to eq(1)
-      delete "/blocks/#{block.id}", params: { id: block.id, format: :json }
-      expect(Block.count).to eq(0)
+    context 'when course is not approved' do
+      it 'does not create an alert, only destroys the block' do
+        expect(Block.count).to eq(1)
+        delete "/blocks/#{block.id}", params: { id: block.id, format: :json }
+        expect(Block.count).to eq(0)
+        expect(Alert.count).to eq(0)
+      end
+    end
+    context 'when course is approved' do
+      it 'create an alert and destroys the block' do
+        course_approved
+        expect(Block.count).to eq(1)
+        delete "/blocks/#{block.id}", params: { id: block.id, format: :json }
+        expect(Block.count).to eq(0)
+        expect(Alert.count).to eq(1)
```

Comment:
This should be a little more specific; no reason not to check for the specific alert type, rather than just the count of the superclass Alert.

Reply:
Looks like you may have missed this comment from last week. ^

Reply:
> This should be a little more specific; no reason not to check for the specific alert type, rather than just the count of the superclass Alert.

@ragesoss what do you mean by this. I wasn't able to understand properly

Reply:
> > This should be a little more specific; no reason not to check for the specific alert type, rather than just the count of the superclass Alert.
> 
> @ragesoss what do you mean by this. I wasn't able to understand properly

Do you want me to check for Alert.type to be equal to CheckTimeline?

Reply:
You could do `CheckTimelineAlert.count` here.

Reply:
@ragesoss instead of that, i did Alert.first type. eq('CheckTimeline') . And it's working you can see my last commit. Should I change it to CheckTimelineAlert.count?

Reply:
This is fine. I think it's a little better not to rely on database order (ie, via `.first`) in tests, but it's a minor thing.